src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -602,6 +612,90 @@ private async Task PerformAppServiceRemoteBuild(Func<Task<Stream>> zipStreamFact
+        }
+
+        public async Task<DeployStatus> PerformFlexDeployment(Site functionApp, Func<Task<Stream>> zipFileFactory, Func<HttpClient, Task<DeployStatus>> deploymentStatusPollTask, IDictionary<string, string> deploymentParameters)
+        {
+            using (var handler = new ProgressMessageHandler(new HttpClientHandler()))
+            using (var client = GetRemoteZipClient(functionApp, handler))
+            using (var request = new HttpRequestMessage(HttpMethod.Post, new Uri(
+                $"api/Deploy/Zip?isAsync=true&author={Environment.MachineName}&Deployer=core_tools&{string.Join("&", deploymentParameters?.Select(kvp => $"{kvp.Key}={kvp.Value}")) ?? string.Empty}", UriKind.Relative)))
+            {
+                ColoredConsole.WriteLine(GetLogMessage("Creating archive for current directory..."));
+
+                request.Headers.IfMatch.Add(EntityTagHeaderValue.Any);
+
+                (var content, var length) = CreateStreamContentZip(await zipFileFactory());
+                request.Content = content;
+                HttpResponseMessage response = await PublishHelper.InvokeLongRunningRequest(client, handler, request, length, "Uploading");
+                await PublishHelper.CheckResponseStatusAsync(response, GetLogMessage("Uploading archive..."));
+
+                // Streaming deployment status for Linux Server Side Build
+                DeployStatus status = await deploymentStatusPollTask(client);
+
+                if (status == DeployStatus.Success)
+                {
+                    // the deployment was successful. Waiting for 60 seconds so that Kudu finishes the sync trigger.
+                    await Task.Delay(TimeSpan.FromSeconds(60));
+
+                    // Checking the function app host status
+                    try
+                    {
+                        await AzureHelper.CheckFunctionHostStatusForFlex(functionApp, AccessToken, ManagementURL);
+                    }
+                    catch (Exception)
+                    {
+                        throw new CliException("Deployment was successful but the app appears to be unhealthy, please check the app logs.");
+                    }
+                    
+                    ColoredConsole.WriteLine(VerboseColor(GetLogMessage("The deployment was successful!")));
+                }
+                else if (status == DeployStatus.Failed)
+                {
+                    throw new CliException("The deployment failed, Please check the printed logs.");
+                }
+                else if (status == DeployStatus.Conflict)
+                {
+                    throw new CliException("Deployment was cancelled, another deployment in progress.");
+                }
+                else if (status == DeployStatus.PartialSuccess)
+                {
+                    ColoredConsole.WriteLine(WarningColor(GetLogMessage("\"Deployment was partially successful, Please check the printed logs.")));
```

Comment:
Same thing as above. We should print the deployment logs since the kudu instance will get recycled and there might be no way for customers to access the logs

Reply:
Logs should already be printed at this stage.